src/ecalc/libraries/libecalc/common/libecalc/output/flow_diagram/EcalcModelMapper.py

```diff
@@ -138,6 +129,56 @@ def _create_compressor_system_diagram(
     return flow_diagrams
 
 
+def _create_compressor_system_diagram(
+    compressor_system: dto.components.CompressorSystem,
+    global_end_date: datetime,
+) -> List[FlowDiagram]:
+    timesteps = _get_timesteps(compressor_system.compressors)
+    return [
+        FlowDiagram(
+            id=compressor_system.id,
+            title=compressor_system.name,
+            start_date=min(timesteps),
+            end_date=max([*timesteps, global_end_date]),
+            nodes=[
+                Node(
+                    id=compressor.name,
+                    title=compressor.name,
+                    type=NodeType.COMPRESSOR,
+                )
+                for compressor in compressor_system.compressors
+            ],
+            edges=[],
+            flows=[],
+        )
+    ]
+
+
+def _create_pump_system_diagram(
+    pump_system: dto.components.PumpSystem,
+    global_end_date: datetime,
+) -> List[FlowDiagram]:
+    timesteps = _get_timesteps(pump_system.pumps)
+    return [
+        FlowDiagram(
+            id=pump_system.id,
+            title=pump_system.name,
+            start_date=min(timesteps),
+            end_date=max([*timesteps, global_end_date]),
+            nodes=[
+                Node(
+                    id=compressor.name,
+                    title=compressor.name,
+                    type=NodeType.COMPRESSOR,
+                )
+                for compressor in pump_system.pumps
```

Comment:
All shown with compressor icons, legacy system looks at the model and displays compressor with turbine different than a compressor train for example.